tests/RobotFramework/libraries/ThinEdgeIO/ThinEdgeIO.py

```diff
@@ -512,6 +518,18 @@ def _assert_mqtt_topic_messages(
 
         return messages
 
+    def log_mqtt_messages(self, topic: str = "#", date_from: Union[datetime, float] = None, **kwargs):
+        items = self.mqtt_match_messages(
+            topic=topic,
+            date_from=date_from,
+            **kwargs,
+        )
+        entries = [
+            f'{item["message"]["tst"].replace("+0000", ""):32} {item["message"]["topic"]:70} {bytes.fromhex(item["payload_hex"]).decode("utf8")}'
```

Comment:
Wouldn't it be safer to pass some error handling scheme like `replace` or `ignore` to that `decode` call so that nothing breaks even if some non-utf8 chars are passed around by some tests?

Reply:
Though I don't think we have any non-utf8 payloads, however I don't want to rule that out in the future.

Reply:
Resovled by 98b5c9180b31769e781b76a3b74e33a4a39dd744